crates/ruma-common/src/http_headers.rs

```diff
@@ -0,0 +1,114 @@
+//! Helpers for HTTP headers.
+
+use std::borrow::Cow;
+
+mod content_disposition;
+mod rfc8187;
+
+pub use self::content_disposition::{
+    ContentDisposition, ContentDispositionParseError, ContentDispositionType, TokenString,
+    TokenStringParseError,
+};
+
+/// Whether the given byte is a [token char].
+///
+/// [token char]: https://datatracker.ietf.org/doc/html/rfc9110#section-5.6.2
+pub const fn is_tchar_byte(b: u8) -> bool {
+    b.is_ascii_alphanumeric()
+        || matches!(
+            b,
+            b'!' | b'#'
+                | b'$'
+                | b'%'
+                | b'&'
+                | b'\''
+                | b'*'
+                | b'+'
+                | b'-'
+                | b'.'
+                | b'^'
+                | b'_'
+                | b'`'
+                | b'|'
+                | b'~'
+        )
+}
+
+/// Whether the given char is a [token char].
+///
+/// [token char]: https://datatracker.ietf.org/doc/html/rfc9110#section-5.6.2
+pub const fn is_tchar(c: char) -> bool {
+    c.is_ascii_alphanumeric()
+        || matches!(
+            c,
+            '!' | '#'
+                | '$'
+                | '%'
+                | '&'
+                | '\''
+                | '*'
+                | '+'
+                | '-'
+                | '.'
+                | '^'
+                | '_'
+                | '`'
+                | '|'
+                | '~'
+        )
+}
```

Comment:
- The first option would make the function non-const. I'm not sure if we care about that, but it would allow to construct a value from a static slice with a compile-time error.
- We could do that, that just changes where the conversion happens. Again that depends of whether we care about being able to use it in const context.
- I don't think it's possible with a `macro_rules!` macro, and it doesn't seem worth it to implement a proc macro just for this.

FTR, std reimplements the functions both time for `char::is_ascii_*` and `u8::is_ascii_*` methods.

Reply:
Oh I have a better idea. We only keep the implementation for `u8` and instead of using `str::chars()`, we use `str::as_bytes()`. Since we only care about ASCII characters, getting bytes should work.

Reply:
Done.